express-notes-api/server.ts

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import express from 'express';
+import { writeFile, readFile } from 'node:fs/promises';
+
+const app = express();
+app.use(express.json());
+
+const json = await readFile('data.json', 'utf8'); // string
+const data = JSON.parse(json); // obj
+
+// Writes data to data.json
+async function writeData() {
+  await writeFile('data.json', JSON.stringify(data, null, 2), 'utf8');
+}
+
+// Reads all notes in the array
+app.get('/api/notes', (req, res) => {
+  let note: any;
+  const notesArray = [];
+  for (note in data.notes) {
+    notesArray.push(data.notes[note]);
+  }
+  res.status(200).json(notesArray);
+});
+
+// Reads the note specified by the inputted id
+app.get('/api/notes/:id', (req, res) => {
+  const { id } = req.params;
+  if (+id < 0) {
+    res.status(400).json({ error: `${id} not a positive integer` });
+  } else if (!(id in data.notes)) {
+    res.status(404).json({ error: `cannot find note with id: ${id}` });
+  } else {
+    res.status(200).json(data.notes[id]);
+  }
+});
+
+// Creates a new note and writes it to data.json
+app.post('/api/notes', async (req, res) => {
+  try {
+    const { content } = req.body;
+    const note = req.body;
+    if (!content) {
+      res.status(400).json({ error: 'content is a required field' });
+    } else if (content) {
```

Comment:
You know at this point that content is defined, so there's no need to check it again. Just a plain `else` is all that's needed.